src/Jackalope/Node.php

```diff
@@ -737,13 +744,13 @@ public function getProperties($nameFilter = null)
     {
         $this->checkState();
 
-        //OPTIMIZE: lazy iterator?
```

Comment:
what you do below does not create a lazy iterator. but probably we don't need one.